src/Tron/Signer.cpp

```diff
@@ -257,10 +279,29 @@ Proto::SigningOutput Signer::sign(const Proto::SigningInput& input) noexcept {
         google::protobuf::Any any;
         any.PackFrom(trigger_smart_contract);
         *contract->mutable_parameter() = any;
+    } else if (input.transaction().has_transfer_trc20_contract()) {
+        auto contract = internal.mutable_raw_data()->add_contract();
+        contract->set_type(protocol::Transaction_Contract_ContractType_TriggerSmartContract);
+
+        auto trigger_smart_contract = to_internal(input.transaction().transfer_trc20_contract());
+        google::protobuf::Any any;
+        any.PackFrom(trigger_smart_contract);
+        *contract->mutable_parameter() = any;
     }
 
-    internal.mutable_raw_data()->set_timestamp(input.transaction().timestamp());
-    internal.mutable_raw_data()->set_expiration(input.transaction().expiration());
+    // Get default timestamp and expiration
+    const uint64_t now = duration_cast< milliseconds >(
+            system_clock::now().time_since_epoch()
+    ).count();
+    const uint64_t timestamp = input.transaction().timestamp() == 0
+            ? now
+            : input.transaction().timestamp();
```

Comment:
it think this logic might be an extra, I would just use what's provided by the client

Reply:
Yes, just use timestamp from app

Reply:
The timestamp is only set if the app doesn't provide it. Same for expiration,